src/foam/nanos/ruler/test/RulerDAOTest.java

```diff
@@ -88,9 +88,15 @@ public void testRuleHistory(X x) {
 
     // wait for async
     try {
-      Thread.sleep(asyncWait + 100);
+      for (int i = 0; i < 2000; i++) {
+        Thread.sleep(asyncWait + 100);
+        ruleHistory = (RuleHistory) ruleHistoryDAO.find(ruleHistory.getId());
+        if ( ruleHistory.getResult().equals("Pending") )
```

Comment:
Since ruleHistory in the pending state, until it is done, we wait for the state to be changed.